eth/peerset.go

```diff
@@ -256,5 +268,6 @@ func (ps *peerSet) close() {
 	for _, p := range ps.peers {
 		p.Disconnect(p2p.DiscQuitting)
 	}
+	close(ps.quitCh)
```

Comment:
Previously, it didn't matter if `close()` was invoked twice, but with this addition, it will cause an error "close on already closed channel", if I remember correctly. So, perhaps change this into
```golang
if !ps.closed { 
   close(ps.quitCh)
}
ps.closed = true
```
It's generally nice if `close()` can be invoked multiple times without problems

Reply:
Follow what you said and handle it as rigorously as possible